packages/core/src/setup.spec.ts

```diff
@@ -0,0 +1,283 @@
+import { describe, it, expect, vi, beforeEach } from "vitest";
+import { setupContract } from "./setup.js";
+import type { Channel } from "amqplib";
+import {
+  defineContract,
+  defineExchange,
+  defineQueue,
+  defineQueueBinding,
+  defineExchangeBinding,
+} from "@amqp-contract/contract";
+
+describe("setupContract", () => {
+  const mockChannel = {
+    assertExchange: vi.fn().mockResolvedValue(undefined),
+    assertQueue: vi.fn().mockResolvedValue(undefined),
+    bindQueue: vi.fn().mockResolvedValue(undefined),
+    bindExchange: vi.fn().mockResolvedValue(undefined),
+  } as unknown as Channel;
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  it("should setup exchanges from contract", async () => {
+    // GIVEN
+    const contract = defineContract({
+      exchanges: {
+        orders: defineExchange("orders", "topic", { durable: true }),
+        notifications: defineExchange("notifications", "fanout", { autoDelete: true }),
+      },
+    });
+
+    // WHEN
+    await setupContract(mockChannel, contract);
+
+    // THEN
+    expect(mockChannel.assertExchange).toHaveBeenCalledTimes(2);
+    expect(mockChannel.assertExchange).toHaveBeenCalledWith("orders", "topic", {
+      durable: true,
+      autoDelete: undefined,
+      internal: undefined,
+      arguments: undefined,
+    });
```

Comment:
always use `toHaveBeenNthCalledWith`, do it everywhere and add it to coding conventions